### PR TITLE
More validation methods

### PR DESCRIPTION
**Pull request by markstory**
In other work I've done, I've found these validation rules to be pretty helpful. 
- The `ascii` rule is good when dealing with legacy databases, or columns where you only want ascii data.
- `isInteger` is a more strict `isNumeric`.
- `utf8` makes it easy to avoid truncated data in MySQL databases that use `utf8` encodings instead of `utf8mb4`. It also lets you prevent emoji if you don't want to accept them.

### CODE REVIEW

File: src/Validation/Validation.php [hunk 1]
```diff
@@ -1098,6 +1098,68 @@ public static function longitude($value, array $options = [])
...
+     * # Options
+     *
+     * - `extended` - Disallow bytes higher within the basic multilingual plane.
+     *   MySQL's older utf8 encoding type does not allow characters above
+     *   the basic multilingual plane. Defaults to false.
+     *
+     * @param string $value The value to check
+     * @return bool
+     */
+    public static function utf8($value, array $options = [])
```
Comment (dereuromark) [inline src/Validation/Validation.php]: isUtf8() maybe?

File: src/Validation/Validation.php [hunk 2]
```diff
@@ -1098,6 +1098,68 @@ public static function longitude($value, array $options = [])
...
 
     /**
+     * Check that the input value is within the ascii byte range.
+     *
+     * This method will reject all non-string values.
+     *
+     * @param string $value The value to check
+     * @return bool
+     */
+    public static function ascii($value)
```
Comment (dereuromark) [inline src/Validation/Validation.php]: isAscii() to be consistent and to know that it returns bool?

Comment (antograssiot) [inline src/Validation/Validation.php]: @dereuromark well I might be wrong but up to now none if rules use the `is` prefix so should rather keep it as is and change `isInteger` to `integer`.

File: src/Validation/Validation.php [hunk 3]
```diff
@@ -1098,6 +1098,68 @@ public static function longitude($value, array $options = [])
...
+     *
+     * This method will reject all non-string values.
+     *
+     * # Options
+     *
+     * - `extended` - Disallow bytes higher within the basic multilingual plane.
+     *   MySQL's older utf8 encoding type does not allow characters above
+     *   the basic multilingual plane. Defaults to false.
+     *
+     * @param string $value The value to check
```
Comment (antograssiot) [inline src/Validation/Validation.php]: CS report a missing
```
@param array $options Options for the check.
```

File: src/Validation/Validation.php (hunk 2 above)
Comment (dereuromark) [inline src/Validation/Validation.php]: Agreed in that case. Even though for 4.0 we might want to think about a better naming scheme in general :)

Comment (markstory) [inline src/Validation/Validation.php]: We can't use `integer` as a method name. Its a keyword.

File: src/Validation/Validation.php (hunk 1 above)
Comment (markstory) [inline src/Validation/Validation.php]: We only use `is` when the methods overlap with reserved words or wrap PHP functions of the same name like `isNumeric`.

File: src/Validation/Validation.php (hunk 2 above)
Comment (dereuromark) [inline src/Validation/Validation.php]: Would it be too much for 3.1 to change all to isX() and deprecate the other ones in 3.x for 4.0?

File: src/Validation/Validation.php (hunk 1 above)
Comment (dereuromark) [inline src/Validation/Validation.php]: In most frameworks and libs as well as in generic coding bool results are expecations of hasX() and isX().
Whereas just x() often means that a return value of some modification of the input is given.
In this case it intuitively sounds like it converts to utf8 and then returns the string.

File: src/Validation/Validation.php (hunk 2 above)
Comment (markstory) [inline src/Validation/Validation.php]: Not very DRY to have every method prefixed with `is` :trollface: :

**Comment by lorenzo** [comment]
this looks good to me :+1:

**Comment by markstory** [comment]
PHPcs build should be fixed now.

File: src/Validation/Validation.php (hunk 1 above)
Comment (markstory) [inline src/Validation/Validation.php]: Fair enough, but every method in this class would then need a prefix which seems excessive to me.

**Comment by markstory** [comment]
I'll update the docs.

**Pull request merged**